httpd-client/lib/project/issue.ts

```diff
@@ -0,0 +1,60 @@
+import type { Comment, ThreadUpdateAction } from "./comment.js";
+import type { ZodSchema } from "zod";
+import { array, boolean, literal, strictObject, string, union } from "zod";
+
+import { commentSchema } from "./comment.js";
+
+export type IssueState =
+  | { status: "open" }
+  | { status: "closed"; reason: "other" | "solved" };
+
+const issueStateSchema = union([
```

Comment:
This file defines schemas but does not define fetching functions. As an alternative you could stick with the hierarchical, resource-oriented approach that you followed for projects and sessions. That is, add a `Client` here that defines issue related method and assign an instance of `Client` to an `issues` property of the project client. Then you would get all issues with `client.project.issues.getAll()`.

I think this approach makes the code more modular because it keeps related things closer together and also more consistent. The downside is that it makes the code also more hierarchical and requires more indirection. (The same observation also applies to commits, comments and patches)

Reply:
I also thought about this. The only reason I split out the types was to keep the `project.ts` file smaller. If we would nest the clients further, where we would put the project ID? Like would we do `client.project.issues.getById(projectId, issueId)`?

Reply:
I hadn’t thought about this. Yes, you could have the project ID as an argument to the issue client methods. But this doesn’t read to nicely. With this in mind, I think the way that it is currently is fine.